rfcs/0156-no-direct-pushes.md

```diff
@@ -0,0 +1,173 @@
+---
+feature: no-direct-pushes
+start-date: 2023-07-21
+author: Silvan Mosberger
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Require pull requests for all Nixpkgs commits.
+
+# Motivation
+[motivation]: #motivation
+
+There are currently 197 people with commit access to Nixpkgs, all of whom can push directly to any branch without a pull request.
+Such pushes generally do not notify anybody and do not trigger CI.
```

Comment:
Strictly speaking that's currently false about not notifying anyone, the false positive bot would also notify about them.

Reply:
That's why I say "generally" here: There's no GitHub mechanism that takes care of notifying other people, there's no builtin way to get notified for new commits linked to https://github.com/NixOS/nixpkgs/issues/118661, but if somebody really wants to they could implement a custom mechanism by e.g. pulling the comments or scanning the repo.